src/encdatavault_fmt_plug.c

```diff
@@ -266,51 +283,78 @@ static int valid(char *ciphertext, struct fmt_main *self)
 err:
 	MEM_FREE(keeptr);
 	return 0;
+
+}
+
+static int valid_md5(char *ciphertext, struct fmt_main *self) {
+	return valid_common(ciphertext, self, 0);
+}
+
+static int valid_pbkdf2(char *ciphertext, struct fmt_main *self) {
+	return valid_common(ciphertext, self, 1);
 }
 
-static void *get_salt(char *ciphertext)
+static void get_salt_common(char *ciphertext, struct salt *cs, int is_pbkdf2)
 {
-	static struct salt cs;
 	int i;
 	char *p = ciphertext, *ctcopy, *keeptr;
 
-	memset(&cs, 0, sizeof(cs));
 	ctcopy = xstrdup(ciphertext);
 	keeptr = ctcopy;
-	ctcopy += FORMAT_TAG_LENGTH;
+
+	if (is_pbkdf2)
+		ctcopy += FORMAT_TAG_PBKDF2_LENGTH;
+	else
+		ctcopy += FORMAT_TAG_MD5_LENGTH;
 	p = strtokm(ctcopy, "$");
-	cs.version = atoi(p);
+	cs->version = atoi(p);
 	p = strtokm(NULL, "$");
-	cs.algo_id = atoi(p);
+	cs->algo_id = atoi(p);
 	p = strtokm(NULL, "$");
 	for (i = 0; i < ENC_NONCE_SIZE; i++)
-		cs.iv[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
+		cs->iv[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
 
 	// Init AES CTR counter
-	cs.iv[ENC_IV_SIZE - 1] = 1;
+	cs->iv[ENC_IV_SIZE - 1] = 1;
 
 	p = strtokm(NULL, "$");
 	for (i = 0; i < ENC_SIG_SIZE; i++)
-		cs.encrypted_data[i + 4] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
+		cs->encrypted_data[i + 4] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
 
-	// Salt and iteration for latest versions
-	if ((cs.version >> 4) == 1) {
+	if (is_pbkdf2) {
+		// Salt and iteration for latest versions
 		p = strtokm(NULL, "$");
-		cs.salt_length = atoi(p);
+		cs->salt_length = atoi(p);
 		p = strtokm(NULL, "$");
-		for (i = 0; i < cs.salt_length; i++)
-			cs.salt[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
+		for (i = 0; i < cs->salt_length; i++)
+			cs->salt[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
 		p = strtokm(NULL, "$");
-		cs.iterations = atoi(p);
+		cs->iterations = atoi(p);
 	}
+
 	// Keychain for version 3
-	if ((cs.version & 0x0f) == 3) {
+	if ((cs->version & 0x0f) == 3) {
 		p = strtokm(NULL, "$");
 		for (i = 0; i < ENC_KEYCHAIN_SIZE; i++)
-			cs.keychain[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
+			cs->keychain[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
 	}
-
 	MEM_FREE(keeptr);
+	return;
+}
+
+static void *get_salt_md5(char *ciphertext) {
+	static struct salt cs;
+
+	memset(&cs, 0, sizeof(cs));
+	get_salt_common(ciphertext, &cs, 0);
+	return (void *)&cs;
+}
+
+static void *get_salt_pbkdf2(char *ciphertext) {
+	static struct salt cs;
+
+	memset(&cs, 0, sizeof(cs));
+	get_salt_common(ciphertext, &cs, 1);
 	return (void *)&cs;
 }
```

Comment:
Please move at least the `memset` back into the common function, and curly braces onto separate lines.